cg/meta/archive/ddn/ddn_data_flow_client.py

```diff
@@ -100,14 +101,15 @@ def auth_header(self) -> dict[str, str]:
             self._refresh_auth_token()
         return {"Authorization": f"Bearer {self.auth_token}"}
 
-    def archive_files(self, files_and_samples: list[FileAndSample]) -> int:
+    def archive_file(self, file_and_sample: FileAndSample) -> int:
         """Archives all files provided, to their corresponding destination, as given by sources
         and destination in TransferData. Returns the job ID of the archiving task."""
         miria_file_data: list[MiriaObject] = self.convert_into_transfer_data(
-            files_and_samples, is_archiving=True
+            [file_and_sample], is_archiving=True
         )
+        metadata: list[dict] = get_metadata(file_and_sample.sample)
```

Comment:
I wonder if this should be here, or passed to the function. What do you think?

Reply:
Currently, all metadata can be retrieved from the FileAndSample object passed to the function, so I feel it would be a bit superfluous to add the metadata on top. If anything changes with the metadata, or if we want the option to override it, it might be good to rework a bit, though. 

We might want to consider adding it to the FileAndSample object or something? If we are certain we would want this metadata for all archive_locations.